src/java/org/apache/cassandra/index/sai/plan/StorageAttachedIndexSearcher.java

```diff
@@ -345,7 +345,7 @@ private boolean isEqualToLastKey(PrimaryKey key)
             // filtered and considered as a result multiple times).
             return lastKey != null &&
                    Objects.equals(lastKey.partitionKey(), key.partitionKey()) &&
-                   Objects.equals(lastKey.clustering(), key.clustering());
+                   (!controller.indexFeatureSet().isRowAware() || Objects.equals(lastKey.clustering(), key.clustering()));
```

Comment:
Am I correct that this works only if for non-row-aware we fetch the data by full partitions, even for the partitions covered by row-aware indexes?